### PR TITLE
GitHub Action for publish

### PR DESCRIPTION
**Pull request by phated**
Relies on #28 

Following https://code.visualstudio.com/api/working-with-extensions/continuous-integration#github-actions - I set up a script for publishing. We'll need to add the publish token in GH token store.

### CODE REVIEW

**Comment by phated** [comment]
Once we know this works, I'll add open-vsix for #25

**Comment by ng-marcus** [comment]
Brilliant, thanks!  I'll try it out today.

**Review by ng-marcus — APPROVED** [review]
This looks good to me.    I guess numbering of releases is based on what's coded into the config?

**Comment by phated** [comment]
Yeah, it comes from the package.json, which we already have to bump before creating tags. I'm going to work on automating that piece too!

**Review by ospencer — APPROVED** [review]
Looks great! Do you already have the secrets or do you need them from me?

**Comment by ospencer** [comment]
@phated Should I open a separate issue for publishing to Open VSX, or can we update this PR to do it as well?

**Comment by phated** [comment]
@ospencer @ng-marcus rebased and tried adding `ovsx publish`. Hopefully that works.

File: editor-extensions/vscode/package.json [hunk 1]
```diff
@@ -1,138 +1,139 @@
...
+	"scripts": {
+		"compile": "tsc -b && node syntaxes/generate-grain-type.js",
+		"watch": "tsc -b -w",
+		"postinstall": "cd client && npm install && cd ../server && npm install && cd ..",
+		"test": "sh ./scripts/e2e.sh",
+		"vscode:prepublish": "npm run compile",
+		"test-compile": "tsc -p ./",
+		"format": "prettier --write .",
+		"check-format": "prettier --check .",
+		"deploy": "vsce publish; ovsx publish"
```
Comment (ospencer) [inline editor-extensions/vscode/package.json:120]: @phated Should this be `vsce publish && ovsx publish` so the CI fails if `vsce publish` fails?

Comment (phated) [reply editor-extensions/vscode/package.json:120]: Doesn't `&&` cause issues on Windows?

Comment (ospencer) [reply editor-extensions/vscode/package.json:120]: A quick Google search claims that `&&` should work fine on Windows. Have you seen issues with it?

Comment (phated) [reply editor-extensions/vscode/package.json:120]: Thought it might be similar to this: https://github.com/grain-lang/grain-lang.org/issues/142 ?

Comment (ospencer) [reply editor-extensions/vscode/package.json:120]: Looks like the issue with that is the use of `&` and `wait`, since you'd need to use different stuff.

Comment (ospencer) [reply editor-extensions/vscode/package.json:120]:
```suggestion
		"deploy": "vsce publish && ovsx publish"
```

**Pull request merged**